lib/BoundaryValueDiffEqAscher/src/ascher.jl

```diff
@@ -315,31 +315,50 @@ function __construct_nlproblem(cache::AscherCache{iip, T}) where {iip, T}
     else
         @closure (z, p) -> @views Φ(cache, z, pt)
     end
+
     lz = reduce(vcat, cache.z)
-    sd = alg.jac_alg.diffmode isa AutoSparse ? SymbolicsSparsityDetection() :
-         NoSparsityDetection()
-    ad = alg.jac_alg.diffmode
-    lossₚ = (iip ? __Fix3 : Base.Fix2)(loss, cache.p)
-    jac_cache = __sparse_jacobian_cache(Val(iip), ad, sd, lossₚ, lz, lz)
-    jac_prototype = init_jacobian(jac_cache)
+    resid_prototype = zero(lz)
+    diffmode = if alg.jac_alg.diffmode isa AutoSparse
+        AutoSparse(alg.jac_alg.diffmode;
+            coloring_algorithm = GreedyColoringAlgorithm(LargestFirst()))
```

Comment:
```suggestion
        AutoSparse(alg.jac_alg.diffmode;
            sparsity_detector = Symbolics.SymbolicsSparsityDetector(),
            coloring_algorithm = GreedyColoringAlgorithm(LargestFirst()))
```
or you can use `TracerSparsityDetector` from SparseConnectivityTracer.jl

Reply:
in any case, without a sparsity detector, you will perform dense AD